### PR TITLE
C++: Exclusion rules for system macros

### PR DESCRIPTION
**Pull request by jbj**
Unwanted results were reported for our JPL Rule 24 queries. Inclusion of system headers with complex macros could lead to unpredictable alerts from these rules.

Fixes https://github.com/github/codeql/issues/6522.

It's hard to write unit tests for these changes, like it was in https://github.com/github/codeql/pull/5085, because there are no system headers in qltests. I tested manually on a vim/vim snapshot and found some examples:
- The `MultipleStmtsPerLine.ql` change removes [this alert for `FD_ISSET`](https://github.com/vim/vim/blob/770ddbe1595f6dab836304203d5ca2e0b069266f/src/channel.c#L4538).
- The `MultipleVarDeclsPerLine.ql` change removes [this alert for `FD_ZERO`](https://github.com/vim/vim/blob/770ddbe1595f6dab836304203d5ca2e0b069266f/src/channel.c#L3440).

### CODE REVIEW

File: cpp/ql/lib/semmle/code/cpp/commons/Exclusions.qll [hunk 1]
```diff
@@ -109,3 +109,39 @@ predicate isFromMacroDefinition(Element e) {
...
+}
+
+/**
+ * Holds if `e` is completely or partially from a _system macro_ definition, as
+ * opposed to being passed in as an argument. A system macro is a macro whose
+ * definition is outside the source directory of the database.
+ * 
+ * If the system macro is invoked through a non-system macro, then this
+ * predicate does not hold. That's a limitation of how macros are represented
+ * in the database.
```
Comment (geoffw0) [inline cpp/ql/lib/semmle/code/cpp/commons/Exclusions.qll:137]: Could you share an example?  (I'd have thought between `getAnExpandedElement()` and `MacroInvocation.getParentInvocation()` we can work this out, I could be wrong)

**Review by geoffw0 — COMMENTED** [review]
> It's hard to write unit tests for these changes, like it was in #5085, because there are no system headers in qltests

`ql\cpp\ql\test\query-tests\Security\CWE\CWE-190\semmle\tainted` implements a system header in a test, though the method is a bit of a fiddle.

**Review by geoffw0 — DISMISSED** [review]
Approved, because neither of my comments needs to block merging this.

File: cpp/ql/lib/semmle/code/cpp/commons/Exclusions.qll (hunk 1 above)
Comment (jbj) [reply cpp/ql/lib/semmle/code/cpp/commons/Exclusions.qll]: I've pushed a commit that removes the claim about db limitations.

**Review by MathiasVP — DISMISSED** [review]

**Comment by jbj** [comment]
I've converted this PR to a draft because we're trying to find out which customers (and which teams) would be affected by this change.

**Comment by jbj** [comment]
We could not get support for this change from all users of these queries, so I'm closing this PR in favour of https://github.com/github/codeql/pull/7014.

**Pull request closed without merging**